src/java/edu/stanford/math/plex4/graph/UndirectedWeightedListGraph.java

```diff
@@ -199,4 +203,12 @@ public int getDegree(int v) {
 		
 		return set.toArray();
 	}
+	
+	public Set<Double> getWeights() {
+		Set<Double> weights = new HashSet<Double>();
+		for (Object x : adjacencySets.getValues())
+			for (Double weight : ((TIntDoubleHashMap) x).getValues())
```

Comment:
This is indeed a little strange. The method adjacencySets.getValues() returns an Object (which is in fact a TIntDoubleHashMap), hence I am forced to cast.